core/src/main/java/org/apache/iceberg/rest/http/RequestResponseSerializers.java

```diff
@@ -0,0 +1,162 @@
+      return Namespace.of(jsonNode.asText().split("\\."));
+    }
+  }
+
+  public static class NamespaceSerializer extends JsonSerializer<Namespace> {
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      gen.writeString(namespace.toString());
+    }
+  }
+
+  public static class TableIdentifierDeserializer extends JsonDeserializer<TableIdentifier> {
+    @Override
+    public TableIdentifier deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return TableIdentifier.parse(jsonNode.asText());
+    }
+  }
+
+  public static class TableIdentifierSerializer extends JsonSerializer<TableIdentifier> {
+    @Override
+    public void serialize(TableIdentifier identifier, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeString(identifier.toString());
+    }
+  }
+
+  public static class SchemaDeserializer extends JsonDeserializer<Schema> {
+    @Override
+    public Schema deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = SchemaParser.fromJson(jsonNode);
+      context.setAttribute("schema", schema);
+      return schema;
+    }
+  }
+
+  public static class SchemaSerializer extends JsonSerializer<Schema> {
+    @Override
+    public void serialize(Schema schema, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      SchemaParser.toJson(schema, gen);
+    }
+  }
+
+  public static class PartitionSpecSerializer extends JsonSerializer<PartitionSpec> {
+    @Override
+    public void serialize(PartitionSpec partitionSpec, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      PartitionSpecParser.toJson(partitionSpec, gen);
+    }
+  }
+
+  public static class PartitionSpecDeserializer extends JsonDeserializer<PartitionSpec> {
+    @Override
+    public PartitionSpec deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+
+      Schema schema = (Schema) context.getAttribute("schema");
+
+      return PartitionSpecParser.fromJson(schema, jsonNode);
+    }
+  }
+
+  public static class SortOrderSerializer extends JsonSerializer<SortOrder> {
+    @Override
+    public void serialize(SortOrder sortOrder, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      SortOrderParser.toJson(sortOrder, gen);
+    }
+  }
+
+  public static class SortOrderDeserializer extends JsonDeserializer<SortOrder> {
+    @Override
+    public SortOrder deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = (Schema) context.getAttribute("schema");
```

Comment:
What writes the schema into the sort order? It looks like these are going to require some context.

Reply:
The `SortOrderParser.fromJson(Schema schema, JsonNode node)` call that returns the final `SortOrder` is what writes the schema into it.

Reply:
That method constructs a `SortOrder` for a given schema that must be passed in. It is on the deserialization side, not the serialization side. For the write side, `SortOrderParser.toJson` produces JSON but doesn't write the schema into it.